src/trace-plugin-interface.js

```diff
@@ -88,6 +94,10 @@ Transaction.prototype.addLabel = function(key, value) {
  * Ends the underlying span. This function should only be called once.
  */
 Transaction.prototype.endSpan = function() {
+  if (this.closed_) {
+    this.agent_.logger.error('A span ended more than once.');
+  }
+  this.closed_ = true;
```

Comment:
I'm hesitant to introduce per state transaction on the transaction objects as they are not unique to the underlying root span.